### DISCUSSION TITLE
What does it mean to "configure IntelliSense"?

### DISCUSSION THREAD

**Discussion by NameOfTheDragon** [microsoft/vscode-makefile-tools#478] [Q&A] [answered]
What is the extension actually doing when it "configures" the project when the build configuration is changed? Is this documented anywhere?

I'm not noticing any detectable change in the environment or settings when this action runs, so I'm wondering
1. What is it supposed to do, and
2. Do I maybe have conflicts with other settings in (for example) the C/C++ extension?

**Comment by andreeis** [comment, marked as answer]
@NameOfTheDragon, during the configure phase, the Makefile Tools extension is reading through a build log and parses compiler commands from which it extracts includes, defines, c/c++ standard and other useful constructs that it passes over to the CppTools extension (which is the IntelliSense brain in VSCode). Makefile Tools also parses for all the targets in the makefile and for linker commands to identify what binaries are built and eventually how are they run, to offer all these in various UI quickPick elements in the left side bar in VSCode. The build log that is read to parse all this is either given via "makefile.buildLog" or the extension invokes "make --dry-run" automatically, which lists all the steps "make" would go through during a real build but "--dry-run" ensures that nothing runs and the state of your project is not altered.

Do you have IntelliSense red squiggles? Includes not found, defines not resolved, any other c/c++ syntax complaints in the "Problems" tab? Conflicts are possible if the Makefile Tools extension is not seen as IntelliSense configuration provider. To make sure, you can use the setting "C_Cpp.default.configurationProvider": "ms-vscode.makefile-tools". Although when that is not defined at all, we detect automatically in the presence of a makefile.

Do you happen to also have CMake build system files within the workspace? Or do you specify other providers or do you set compileCommands anywhere?

**Comment by andreeis** [comment]
@NameOfTheDragon, we are cleaning up our tickets in the repository and closing the ones which have been addressed or which are too old without any recent feedback. Let us know if you need more assistance or clarity with this question. I plan to close it.

**Reply by guhcampos** [reply]
I was about to write a question which I think is related to this. I got confused by the VS Code prompt to configure the extension, and reading through the docs didn't enlighten me any better.

I then noticed the `Makefile` view in VSCdode has these `Unset` targets for `Configuration`, and was trying to find if there's an expected `Makefile` structure the extension wants to have, but I could not find any. I'm still confused as to how is this supposed to be configured and what exactly it wants to execute every time a new project is added. I believe a bit of clarification in the docs could go a long way.

Maybe provide an example Makefile and go through the steps the extension goes through by default? My makefiles are not very standard, I'm arguably overloading `make` to setup local development tools for Kubernetes services, so I'm fine with having to manually configure the extension if needed, but having some guidance would help. Could be as simple as "Your makefile should have a target named `potato` so we can pick it up and do `XYZ` or something like that.
